src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruitName;
+    private int transactionQuantity;
+    public FruitTransaction(Operation operation, String fruitName, int transactionQuantity) {
+        this.operation = operation;
+        this.fruitName = fruitName;
+        this.transactionQuantity = transactionQuantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+    }
```

Comment:
Consider moving the `Operation` enum to a separate file to make the code more organized and to follow the checklist point about keeping logic in appropriate packages.